### PR TITLE
Remove SwiftOnoneSupport & Comments

### LINKED ISSUE #5: I would like to ignore documentation
No seriously, can we just never write documentation :trollface: 

Usage

```
public-api-generator --no-docs
```

Justification

Depending on the use case it might be valuable to have a much more simplified look at the interface, and judge more on the naming of variables and functions rather than reading all the documentation. This is obviously dependent on individual use-cases, and shouldn't be on by default

### LINKED ISSUE #6: Remove Swift library from headers
`SwiftOnoneSupport`

### PR DESCRIPTION
**Pull request by ollieatkinson**
Fix #5
Fix #6

Todo:
- Write tests for `GenerateCommand.swift`
- Ensure we are happy with regex replace for `RemoveComments.swift`. Currently this works fine, but deletes new lines between code, potentially consider new regex or figure out how to strip leading whitespace when using `sourcekitten`.

### CODE REVIEW

File: Sources/GuiseFramework/APIGenerator.swift [hunk 1]
```diff
@@ -1,10 +1,24 @@
 import Foundation
 import SourceKittenFramework
 
+protocol SourceKittenRequest {
```
Comment (paulsamuels) [inline Sources/GuiseFramework/APIGenerator.swift]: Is this change related? It looks like this is the start of making `APIGenerator` testable but need further work

Comment (ollieatkinson) [reply Sources/GuiseFramework/APIGenerator.swift]: No, this needs moving into it's own request - good spot!

File: Sources/GuiseFramework/TextProcessors/RemoveComments.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+struct RemoveComments: TextProcessor {
+
+  func process(input: String) throws -> String {
+
+    return input.split(separator: "\n").filter({ line in
+      line.range(of: " *\\/\\/\\/", options: .regularExpression) == nil
```
Comment (paulsamuels) [inline Sources/GuiseFramework/TextProcessors/RemoveComments.swift]: - Could tidy the regex up a tiny bit with " *\\/{3}"
- It looks like the `split` that is dropping the new lines - can add `omittingEmptySubsequences: false`

File: Tests/GuiseFrameworkTests/TextProcessors/RemoveCommentsTests.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+@testable import GuiseFramework
+
+class RemoveCommentsTests: XCTestCase {
+  
+  func testStripComments() throws {
+    
+    let input = """
+/// This is a doc comment for my structure
+struct MyStructure {
+  /// This is a doc comment for my property
```
Comment (paulsamuels) [inline Tests/GuiseFrameworkTests/TextProcessors/RemoveCommentsTests.swift:11]: Is it worth adding a new line into this test case so we can catch if new lines are accidentally stripped?

Comment (ollieatkinson) [reply Tests/GuiseFrameworkTests/TextProcessors/RemoveCommentsTests.swift:11]: Yep! Sounds like a good idea

File: Tests/GuiseFrameworkTests/TextProcessors/RemoveSwiftOnoneSupportTests.swift [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import XCTest
+@testable import GuiseFramework
+
+class RemoveSwiftOnoneSupportTests: XCTestCase {
+  
+  func testRemoveIfPresent() throws {
+    
+    let systemUnderTest = RemoveSwiftOnoneSupport()
+    
+    XCTAssertEqual(try systemUnderTest.process(input: "\nimport SwiftOnoneSupport\nimport Foundation"), "\nimport Foundation", "SwiftOnoneSupport should be removed, but it was not.")
```
Comment (paulsamuels) [inline Tests/GuiseFrameworkTests/TextProcessors/RemoveSwiftOnoneSupportTests.swift]: This test case is not realistic as the file might not start with a new line. I guess that also raises that this could potentially not delete the import if it was the first line for some reason. Should it be `import SwiftOnoneSupport\n` instead?

**Review by paulsamuels — APPROVED** [review]
One remaining question but looks good 🚗

File: Sources/GuiseFramework/CLI/GenerateCommand.swift [hunk 5]
```diff
@@ -77,6 +81,21 @@ struct GenerateCommand: CommandProtocol {
       }
     }
     
+
```
Comment (Sherlouk) [inline Sources/GuiseFramework/CLI/GenerateCommand.swift]: nit: double whitespace

File: Sources/GuiseFramework/CLI/GenerateCommand.swift [hunk 6]
```diff
@@ -77,6 +81,21 @@ struct GenerateCommand: CommandProtocol {
...
+    if options.noDocumentation {
+      textProcessors.append(RemoveComments())
+    }
+    
+    do {
+      source = try textProcessors.reduce(source, { try $1.process(input: $0) })
+    } catch {
+      return .failure(.unexpectedError(error: error))
+    }
+
```
Comment (Sherlouk) [inline Sources/GuiseFramework/CLI/GenerateCommand.swift:97]: General comment, not a blocker for this but if this function continues to expand with additional functionality then we should probably think about breaking it down a little. You've already got comments breaking it into logical chunks!
